### ISSUE TITLE
Unexpected differences between remote.SSH.useLocalServer false and true

### ISSUE THREAD

**Issue by bersbersbers** [microsoft/vscode-remote-release#4492]
- VSCode Version: 1.53.2
- Local OS Version: Windows 10
- Remote OS Version: OpenSUSE Leap 15.2
- Remote Extension/Connection Type: SSH

`"remote.SSH.useLocalServer": false` works fine, so I'd expect `true` to also work. Here's what different:

1. My local username is Firstname, my remote username ist lastname. I have my `user lastname` set in my SSH config, so I can connect using `ssh hostname`. However, the local server seems to connect to `Firstname@host`, which is incorrect. This can be worked around by connecting to `lastname@host` in VS Code, but that should not be necessary.

2. I usually set no specific SSH path in VS Code - global `PATH` is sufficient. However, with local server, it does not seem to like my default `ssh.exe` on the `PATH` and uses the Git one instead:
```
[06:38:55.343] Checking ssh with "ssh -V"
[06:38:55.402] > OpenSSH_for_Windows_8.1p1, LibreSSL 2.9.2

[06:38:55.407] Checking ssh with "C:\WINDOWS\System32\OpenSSH\ssh.exe -V"
[06:38:55.409] Got error from ssh: spawn C:\WINDOWS\System32\OpenSSH\ssh.exe ENOENT
[06:38:55.410] Checking ssh with "C:\Program Files\Git\usr\bin\ssh.exe -V"
[06:38:55.464] > OpenSSH_8.4p1, OpenSSL 1.1.1i  8 Dec 2020

[06:38:55.476] Using SSH config file "C:\Users\Firstname\.ssh\config"
```
I don't have an agent configured for my Git `ssh.exe` (I never asked for that to be installed), so it does not connect. This can be worked around using `"remote.SSH.path": "C:\\Tools\\OpenSSH\\ssh.exe"`, but that should not be necessary.

TLDR: `"remote.SSH.useLocalServer": true` can lead to wrong username and ssh executable being used when taking a working setup with `"remote.SSH.useLocalServer": false` as a baseline.

**Comment by bersbersbers** [comment]
2 may be intended, as explained by the quote in https://github.com/microsoft/vscode-remote-release/issues/1521#issue-500567109

**Comment by bersbersbers** [comment]
1 is discussed at length in https://github.com/microsoft/vscode-remote-release/issues/2512, but I fail to find a solution other than disabling `remote.SSH.useLocalServer`.

**Comment by roblourens** [comment]
Yeah, this is tough on windows right now. There is a bug in the windows OpenSSH build that can cause issues using useLocalServer, so I try to prefer git SSH, but that can cause other issues, this is basically why useLocalServer is off by default on Windows.

Setting `remote.SSH.path` is my recommended workaround for now if you want to want to use Windows OpenSSH with useLocalServer, sounds like that is working for you. The main issue is that if there is an auth prompt, we don't get the real auth message from ssh. But a windows update in the next couple months will have a fix for that at least.
